File: moviejava/app/src/main/java/com/example/moviejava/extentions/ViewDiffUtil.java
package com.example.moviejava.extentions;

import androidx.recyclerview.widget.DiffUtil;

import com.example.moviejava.model.Results;

import java.util.List;


public class ViewDiffUtil extends DiffUtil.Callback {

    private List<Results> oldList;
    private List<Results> newList;

    public ViewDiffUtil(List<Results> oldList, List<Results> newList) {
        this.oldList = oldList;
        this.newList = newList;
    }

    @Override
    public int getOldListSize() {
        return oldList.size();
    }

    @Override
    public int getNewListSize() {
        return newList.size();
    }

    @Override
    public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
        return oldList.get(oldItemPosition).id == newList.get(newItemPosition).id;
    }

    @Override
    public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
        return oldList.get(oldItemPosition)==newList.get(newItemPosition);
    }
}

File: movie/app/src/main/java/com/example/movie/extensions/ViewDiffUtil.kt
package com.example.movie.extensions

import androidx.recyclerview.widget.DiffUtil
import com.example.movie.model.Results

class ViewDiffUtil(
    private val oldList: List<Results>,
    private val newList: List<Results>
) : DiffUtil.Callback() {

    override fun getOldListSize(): Int = oldList.size


    override fun getNewListSize(): Int = newList.size

    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return oldList[oldItemPosition].id == newList[newItemPosition].id
    }

    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return oldList[oldItemPosition] == newList[newItemPosition]

    }
}